src/lava/magma/compiler/builders/interfaces.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+from abc import ABC, abstractmethod
+
+import typing as ty
+
+from lava.magma.compiler.channels.interfaces import AbstractCspPort
+if ty.TYPE_CHECKING:
+    from lava.magma.core.model.model import AbstractProcessModel
+
+
+class AbstractProcessBuilder(ABC):
+    @abstractmethod
+    def set_csp_ports(self, csp_ports: ty.List[AbstractCspPort]):
+        pass
+
+    @property
+    @abstractmethod
+    def proc_model(self) -> "AbstractProcessModel":
+        pass
+
+
+class AbstractRuntimeServiceBuilder(ABC):
```

Comment:
+docstring
+type annotations for __init__ arguments

Reply:
ACK